easypost/services/base_service.py

```diff
@@ -0,0 +1,101 @@
+import re
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+)
+
+from easypost.easypost_object import convert_to_easypost_object
+from easypost.error import Error
+from easypost.requestor import (
+    RequestMethod,
+    Requestor,
+)
+
+
+class BaseService:
+    """The base service that all other services inherit."""
+
+    def __init__(self, client):
+        self.client = client
+
+    def snakecase_name(self, class_name) -> str:
+        """Return the class name as snake_case."""
+        return re.sub(r"(?<!^)(?=[A-Z])", "_", class_name).lower()
+
+    def class_url(self, class_name) -> str:
+        """Generate a URL based on class name."""
+        if class_name[-1:] in ("s", "h"):
+            return f"/{class_name}es"
+        else:
+            return f"/{class_name}s"
+
+    def instance_url(self, class_name, id) -> str:
+        """Generate an instance URL based on the ID of the object."""
+        if not id:
+            raise Error(f"{class_name} instance has invalid ID: {id}")
+
+        class_url = self.class_url(class_name)
+
+        return f"{class_url}/{id}"
```

Comment:
Issue: missing type hints on these functions

Reply:
Will fix all type hints in a follow up PR